pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -653,18 +653,48 @@ fi
 # If we're not just building, then make the new configuration the boot
 # default and/or activate it now.
 if [[ "$action" = switch || "$action" = boot || "$action" = test || "$action" = dry-activate ]]; then
+    # Using systemd-run here to protect against PTY failures/network
+    # disconnections during rebuild.
+    # See: https://github.com/NixOS/nixpkgs/issues/39118
+    cmd=(
+        "systemd-run"
+        "-E" "LOCALE_ARCHIVE" # Will be set to new value early in switch-to-configuration script, but interpreter starts out with old value
+        "--collect"
+        "--no-ask-password"
+        "--pty"
+        "--quiet"
+        "--same-dir"
+        "--service-type=exec"
+        "--unit=nixos-rebuild-switch-to-configuration"
+        "--wait"
+    )
+    # Check if we have a working systemd-run. In chroot environments we may have
+    # a non-working systemd, so we fallback to not using systemd-run.
+    # You may also want to explicitly set NIXOS_IGNORE_SYSTEMD_RUN environment
+    # variable, since systemd-run runs inside an isolated environment and
+    # this may break some post-switch scripts. However keep in mind that this
+    # may be dangerous in remote access (e.g. SSH).
+    if [[ -n "$NIXOS_IGNORE_SYSTEMD_RUN" ]]; then
+        log "warning: skipping systemd-run since NIXOS_IGNORE_SYSTEMD_RUN is set. This environment variable will be ignored in the future"
+        cmd=()
+    elif ! targetHostCmd "${cmd[@]}" true &>/dev/null; then
+        logVerbose "Skipping systemd-run to switch configuration since it is not working in target host."
+        cmd=()
```

Comment:
In this case we should clean the environment, for consistency with normal deployments.

`NIXOS_IGNORE_SYSTEMD_RUN` should perhaps be `NIXOS_SWITCH_USE_DIRTY_ENV`.

Reply:
Done.

Reply:
I think we should clean the environment for consistency with `systemd-run`.
`switch-to-configuration` will set `PATH`.

A first approximation would be

```suggestion
        cmd=("env" "-i" "LOCALE_ARCHIVE=$LOCALE_ARCHIVE")
```

However, I don't know what `systemd-run` preserves and/or loads.

Reply:
Done.

Reply:
fwiw unless you've done `systemctl import-environment` before. For instance

```
$ export FNORD=bar
$ sudo -E systemctl import-environment FNORD
$ sudo systemd-run -t --wait env 2>/dev/null | grep FNORD
FNORD=bar
```

Also, `import-environment` w/o args imports the full environment.